met-web/src/components/layout/SideNav/SideNav.tsx

```diff
@@ -28,40 +40,142 @@ const DrawerBox = () => {
+                            filter: 'brightness(96%)',
+                        },
+                        '&:active': {
+                            filter: 'brightness(92%)',
+                        },
+                    }}
+                >
+                    <ListItemButton
+                        data-testid={`SideNav/${route.name}-button`}
+                        onClick={() => {
+                            navigate(route.path);
+                            setOpen(false);
+                        }}
+                    >
+                        {handleFontAwesomeIcon(route, itemType)}
+                        <MetHeader4
+                            style={{
+                                color: 'selected' === itemType ? Palette.primary.light : Palette.text.primary,
+                                fontWeight: 'selected' === itemType ? 'bold' : '500',
+                                fontSize: '1rem',
+                            }}
+                        >
+                            {route.name}
+                            <When condition={currentBaseRoute === route.base}>
+                                <span style={{ position: 'absolute', right: '1rem' }}>
+                                    <FontAwesomeIcon icon={faCheck} />
+                                </span>
+                            </When>
+                        </MetHeader4>
+                    </ListItemButton>
+                </ListItem>
+                <Divider sx={{ backgroundColor: colors.surface.gray[30] }} />
+                <When condition={'User Admin' === route.name}>
+                    <UserGuideNav />
+                </When>
+            </>
+        );
+    };
+
+    const handleFontAwesomeIcon = (route: Route, itemType: string) => {
+        const fontAwesomeStyles = {
+            fontSize: '1.1rem',
+            color: 'selected' === itemType ? Palette.primary.light : Palette.text.primary,
+            paddingRight: '0.75rem',
+            width: '1.1rem',
+        };
+        const iconAssignments: IconAssignments = {
+            Home: faHouse,
+            Engagements: faPeopleArrows,
+            Surveys: faSquarePollHorizontal,
+            Metadata: faTags,
+            Languages: faGlobe,
+            'User Admin': faUserGear,
+            'Tenant Admin': faHouseUser,
+            'MET Feedback': faMessagePen,
+        };
+        // If the route name doesn't match an expected page title, return the broken link icon
+        const possiblePageTitles: string[] = Object.keys(iconAssignments).filter((key) => key.includes(route.name));
+        if (0 === possiblePageTitles.length) {
+            return <FontAwesomeIcon icon={faLinkSlash} style={fontAwesomeStyles} />;
+        }
+        // Otherwise return the icon for the icon assignments key name that matches the route name.
+        return <FontAwesomeIcon icon={iconAssignments[route.name]} style={fontAwesomeStyles} />;
+    };
+
+    const CloseButton = () => {
```

Comment:
Sonarcloud is suggesting (and I tend to agree) that this component definition should be extracted from the DrawerBox component and have its onClick behaviour passed in. Otherwise, just create it inline and use `<When>`

Reply:
No problem, I changed this so that the component is not within the DrawerBox.